src/agent.ts

```diff
@@ -12,52 +12,81 @@ import {
   FindingType,
 } from "forta-agent";
 
-export const ERC20_TRANSFER_EVENT =
-  "event Transfer(address indexed from, address indexed to, uint256 value)";
-export const TETHER_ADDRESS = "0xdAC17F958D2ee523a2206206994597C13D831ec7";
-export const TETHER_DECIMALS = 6;
+import { NetworkManager } from "forta-agent-tools";
+import ethers from "forta-agent";
+
+import { networkData, NetworkData } from "./network";
+import {
+  methods,
+  fortaRegistryAddress,
+  nethermindDeployAddress,
+} from "./constants";
+
+const networkManager = new NetworkManager(networkData); //global var
+
+// export const ERC20_TRANSFER_EVENT =
+//   "event Transfer(address indexed from, address indexed to, uint256 value)";
+// export const TETHER_ADDRESS = "0xdAC17F958D2ee523a2206206994597C13D831ec7";
+// export const TETHER_DECIMALS = 6;
 let findingsCount = 0;
 
-const handleTransaction: HandleTransaction = async (
-  txEvent: TransactionEvent
-) => {
-  const findings: Finding[] = [];
-
-  // limiting this agent to emit only 5 findings so that the alert feed is not spammed
-  if (findingsCount >= 5) return findings;
-
-  // filter the transaction logs for Tether transfer events
-  const tetherTransferEvents = txEvent.filterLog(
-    ERC20_TRANSFER_EVENT,
-    TETHER_ADDRESS
-  );
-
-  tetherTransferEvents.forEach((transferEvent) => {
-    // extract transfer event arguments
-    const { to, from, value } = transferEvent.args;
-    // shift decimals of transfer value
-    const normalizedValue = value.div(10 ** TETHER_DECIMALS);
-
-    // if more than 10,000 Tether were transferred, report it
-    if (normalizedValue.gt(10000)) {
-      findings.push(
-        Finding.fromObject({
-          name: "High Tether Transfer",
-          description: `High amount of USDT transferred: ${normalizedValue}`,
-          alertId: "FORTA-1",
-          severity: FindingSeverity.Low,
-          type: FindingType.Info,
-          metadata: {
-            to,
-            from,
-          },
-        })
-      );
-      findingsCount++;
-    }
-  });
-
-  return findings;
+export const provideHandleTransaction = (
+  methods: string[],
+  nethermindDeployAddress: string,
+  fortaRegistryAddress: string
+): HandleTransaction => {
+  return async (txEvent: TransactionEvent) => {
+    const findings: Finding[] = [];
+
+    // limiting this agent to emit only 5 findings so that the alert feed is not spammed
+    if (findingsCount >= 5) return findings;
+
+    // filter TransactionEvent by methods for bot deployments and updates agents on Forta registry smart contract
+    const functionCalls = txEvent.filterFunction(methods, fortaRegistryAddress);
+
+    functionCalls.forEach((call) => {
+      // extract transfer event arguments
+      const { agentId, metadata, chainIds } = call.args;
+
+      if (call.name === "createAgent") {
+        findings.push(
+          Finding.fromObject({
+            name: "Nethermind Forta Bot Create Agent",
```

Comment:
```suggestion
            name: "Nethermind Forta Bot Created",
```
Suggestion for a slightly simpler name. Applicable to the other alert.